docs/modules/tasks.md

```diff
@@ -0,0 +1,231 @@
+
+## Usage
+
+### Specifying
+Every `Environment` instance includes a task, defined by its config that is passed to the environment constructor via the `task` key.
+This is expected to be a dictionary of relevant keyword arguments, specifying the desired task configuration to be created (e.g. reward type and weights, hyperparameters for reset behavior, etc).
+The `type` key is required and specifies the desired task class. Additional keys can be specified and will be passed directly to the specific task class constructor.
+An example of a task configuration is shown below in `.yaml` form:
+
+??? code "point_nav_example.yaml"
+    ``` yaml linenums="1"
+    task:
+      type: PointNavigationTask
+      robot_idn: 0
+      floor: 0
+      initial_pos: null
+      initial_quat: null
+      goal_pos: null
+      goal_tolerance: 0.36    # turtlebot bodywidth
+      goal_in_polar: false
+      path_range: [1.0, 10.0]
+      visualize_goal: true
+      visualize_path: false
+      n_vis_waypoints: 25
+      reward_type: geodesic
+      termination_config:
+        max_collisions: 500
+        max_steps: 500
+        fall_height: 0.03
+      reward_config:
+        r_potential: 1.0
+        r_collision: 0.1
+        r_pointgoal: 10.0
+    ```
+
+### Runtime
+
+`Environment` instance has a `task` attribute that is an instance of the specified task class.
+Internally, `Environment`'s `reset` method will call the task's `reset` method, `step` method will call the task's `step` method, and the `get_obs` method will call the task's `get_obs` method.
+
+## Types
+**`OmniGibson`** currently supports 5 types of tasks, 7 types of termination conditions, and 5 types of reward functions.
+
+### `Task`
+
+<table markdown="span">
+    <tr>
+        <td valign="top">
+            [**`DummyTask`**](../reference/tasks/dummy_task.html)<br><br>
+            Dummy task with trivial implementations.
+            <ul>
+                <li>`termination_conditions`: empty dict.</li>
+                <li>`reward_functions`: empty dict.</li>
+                <li>`_get_obs()`: empty dict.</li>
+                <li>`_reset_scene()`: default.</li>
+                <li>`_reset_agent()`: default.</li>
+            </ul>
+        </td>
+    </tr>
+    <tr>
+        <td valign="top">
+            [**`PointNavigationTask`**](../reference/tasks/point_navigation_task.html)<br><br>
+            PointGoal navigation task with fixed / randomized initial pose and goal location.
+            <ul>
+                <li>`termination_conditions`: `MaxCollision`, `Timeout`, `PointGoal`.</li>
+                <li>`reward_functions`: `PotentialReward`, `CollisionReward`, `PointGoalReward`.</li>
+                <li>`_get_obs()`: returns relative xy position to the goal, and the agent's current linear and angular velicities.</li>
```

Comment:
nit: velicities -> velocities (also elsewhere).

Can we also specify the dimension size of these observations?

Reply:
fixed the typo. re dimension, I think the reader should just look at the docstring of the code. This module documentation is meant to provide some high-level guidance. But of course if you disagree, we can discuss and fix later!